src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -568,6 +569,48 @@ def enable_hdi_monitoring(cmd, client, resource_group_name, cluster_name, worksp
         monitor_request_parameter)
 
 
+# pylint: disable=unused-argument
+def enable_hdi_azure_monitor(cmd, client, resource_group_name, cluster_name, workspace, primary_key=None,
+                             workspace_type='resource_id', no_validation_timeout=False):
+    from azure.mgmt.hdinsight.models import AzureMonitorRequest
+    from msrestazure.tools import parse_resource_id
+    from ._client_factory import cf_log_analytics
+
+    if workspace_type != 'resource_id' and not primary_key:
+        raise CLIError('primary key is required when workspace ID is provided')
```

Comment:
CLIError is deprecated. Please follow the [guide line](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#available-error-types) and use a detailed error type.

Reply:
> CLIError is deprecated. Please follow the [guide line](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#available-error-types) and use a detailed error type.

Hi @kairu-ms , I have fixed this line. And I also checked the other CLIError usages, For the other CLIError I can not find a correct replacement so keep the other CLIError usages.
Could you please help review again?